File: backend/src/user/entities/coach.entity.ts
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  OneToMany,
  ManyToOne,
  JoinTable,
  ChildEntity,
} from 'typeorm';

import { CoachRole } from './../models/coach-role.enum';

import { User } from './user.entity';
import { Client } from './client.entity';
import { Organization } from '@app/organization/entities/organization.entity';
import { Program } from '@app/program/entities/program.entity';
import { Workout } from '@app/workout/entities/workout.entity';

@ChildEntity()
export class Coach extends User {
  @Column({
    type: 'enum',
    enum: CoachRole,
  })
  role: CoachRole;

  @JoinTable()
  @OneToMany(
    type => Client,
    client => client.coach,
  )
  clients: Client[];

  @JoinTable()
  @OneToMany(
    type => Program,
    program => program.coach,
  )
  programs: Program[];

  @JoinTable()
  @OneToMany(
    type => Workout,
    workout => workout.coach,
  )
  workouts: Workout[];

  @ManyToOne(
    type => Organization,
    organization => organization.coaches,
  )
  organization: Organization;
}

File: mobile_app/lib/entities/coach.entity.dart
import 'package:mobile_app/entities/organization.entity.dart';
import 'package:mobile_app/entities/user.entity.dart';

import 'client.entity.dart';
import 'package:json_annotation/json_annotation.dart';

part 'coach.entity.g.dart';

@JsonSerializable()
class Coach extends User {
  String role;
  List<Client> clients;
  Organization organization;

  Coach({this.role, this.clients, this.organization});

  factory Coach.fromJson(Map<String, dynamic> json) => _$CoachFromJson(json);
  Map<String, dynamic> toJson() => _$CoachToJson(this);
}
